### PR TITLE
Add data transformer examples to docs

### LINKED ISSUE #2557: Update `sncast` docs to use data transformer instead of serialized calldata as a default exampl
### Which components does the task require to be changed? (think hard pls)

sncast

### Description

Once data transformer has proven itself to be working correctly we could update the documentation so default way of passing arguments is shown as data transformation. 

Current way of passing already serialized calldata could be documented as an option.

### PR DESCRIPTION
**Pull request by kkawula**
Closes #2557

## Introduced changes

-

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: docs/src/starknet/index.md [hunk 1]
```diff
@@ -51,14 +50,16 @@ response: [0x0]
 > In the above example we supply `sncast` with `--account` and `--url` flags. If `snfoundry.toml` is present, and have these properties set, values provided using these flags will override values from `snfoundry.toml`. Learn more about `snfoundry.toml` configuration [here](../projects/configuration.md#sncast).
 
 
-### Calldata
+### Arguments and Calldata
```
Comment (cptartur) [inline docs/src/starknet/index.md:53]: How about we change this section to `### Arguments` and explain only how to use `--arguments` flag.

Below, we can add a section `### Using Serialized Calldata` and keep the docs for `--calldata`.

File: docs/src/starknet/index.md [hunk 2]
```diff
@@ -51,14 +50,16 @@ response: [0x0]
...
 
 Some `sncast` commands (namely `call`, `deploy` and `invoke`) allow passing *calldata* - a series of arguments to perform an action with on blockchain.
 
-In the example above we called a function with an argument: `0x0`, passed using `--calldata` flag.
+In the example above we called a function with an argument: `'"Charizard"'`, passed using `--arguments` flag, but it can
+be also obtained by passing a serialized form, in this example equivalent to passing `--calldata` with the following value: `0x0 0x43686172697a617264 0x9`.
 
 Please note the notation of the argument. The default way of passing calldata is a list of hexadecimally encoded field elements - the *serialized* calldata.
 To obtain the serialized form of the wished data, one must write a Cairo program calling `Serde::serialize` on subsequent arguments and displaying the results.
```
Comment (cptartur) [inline docs/src/starknet/index.md:61]: This is outdated, serialized is no longer default - there is no default in code at all.

**Review by cptartur — CHANGES_REQUESTED** [review]

File: docs/src/starknet/index.md [hunk 3]
```diff
@@ -51,21 +50,29 @@ response: [0x0]
...
+
+### Using Serialized Calldata
+
+Equivalent to the previous example, the same result can be obtained by passing serialized calldata - a list of hexadecimal encoded field elements.
+
+In this example equivalent to passing `--calldata` with the following value: `0x0 0x43686172697a617264 0x9`.
+
+To obtain the serialized form of the wished data, one must write a Cairo program calling `Serde::serialize` on subsequent arguments and displaying the results.
+Read more about it in the [Cairo documentation](https://book.cairo-lang.org/appendix-03-derivable-traits.html?highlight=seri#serializing-with-serde).
+
```
Comment (franciszekjob) [inline docs/src/starknet/index.md:75]:
```suggestion

### Using Serialized Calldata

The same result can be achieved by passing serialized calldata, which is a list of hexadecimal-encoded field elements.

For example, this is equivalent to using the --calldata option with the following value: 0x0 0x43686172697a617264  0x9.

To obtain the serialized form of the wished data, you can write a Cairo program that calls `Serde::serialize` on subsequent arguments and displays the results.

Read more about it in the [Cairo documentation](https://book.cairo-lang.org/appendix-03-derivable-traits.html?highlight=seri#serializing-with-serde).

```

**Review by franciszekjob — COMMENTED** [review]
These examples (snippets + outputs) rely on example contract(s) present in our code, right?

File: docs/src/starknet/index.md [hunk 4]
```diff
@@ -51,21 +50,30 @@ response: [0x0]
...
 > ⚠️ **Warning**
 > Cast will not verify the serialized calldata. Any errors caused by passing improper calldata in a serialized form will originate from the network.
 > Basic static analysis is possible only when passing expressions - see [calldata transformation](./calldata-transformation.md).
 
+
+### Using Serialized Calldata
+
+The same result can be achieved by passing serialized calldata, which is a list of hexadecimal-encoded field elements.
+
+For example, this is equivalent to using the --calldata option with the following value: 0x0 0x43686172697a617264  0x9.
```
Comment (cptartur) [inline docs/src/starknet/index.md:71]:
```suggestion
For example, this is equivalent to using the --calldata option with the following value: 0x0 0x43686172697a617264 0x9.
```

File: docs/src/starknet/index.md [hunk 5]
```diff
@@ -51,21 +50,30 @@ response: [0x0]
...
+
+To obtain the serialized form of the wished data, you can write a Cairo program that calls `Serde::serialize` on subsequent arguments and displays the results.
```
Comment (cptartur) [inline docs/src/starknet/index.md:73]: Kind of resolves https://github.com/foundry-rs/starknet-foundry/issues/2650 but I think we might want to expand this in a separate PR anyway.

**Review by cptartur — APPROVED** [review]

**Pull request merged**